app/controllers/instructor/sections_controller.rb

```diff
@@ -13,7 +13,7 @@ def create
 	end
 
 	def update
-		current_course.update_attributes(section_params)
+		current_section.update_attributes(section_params)
```

Comment:
The update action for the `sections` controller, should update the `section` -> not the course.